pkg/pxc/app/statefulset/haproxy.go

```diff
@@ -0,0 +1,218 @@
+			{
+				Name:      "haproxy-cfg",
+				MountPath: "/etc/haproxy/pxc",
+			},
+			{
+				Name:      "haproxy-auto",
+				MountPath: "/etc/haproxy-auto/",
+			},
+		},
+		Env: []corev1.EnvVar{
+			{
+				Name:  "POD_NAME",
+				Value: c.sfs.ObjectMeta.Name,
+			},
+			{
+				Name:  "POD_NAMESPACE",
+				Value: c.sfs.ObjectMeta.Namespace,
+			},
+			{
+				Name:  "PXC_SERVICE",
+				Value: c.labels["app.kubernetes.io/instance"] + "-" + "pxc",
+			},
+			{
+				Name: "MONITOR_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: app.SecretKeySelector("my-cluster-secrets", "monitor"),
+				},
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			FailureThreshold:    3,
+			SuccessThreshold:    1,
+			TimeoutSeconds:      1,
+			PeriodSeconds:       10,
+			InitialDelaySeconds: initDelay,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path:   "/healthz",
+					Port:   intstr.FromInt(1024),
+					Scheme: "HTTP",
+				},
+			},
+		},
+		SecurityContext: spec.ContainerSecurityContext,
+	}
+
+	res, err := app.CreateResources(spec.Resources)
+	if err != nil {
+		return appc, fmt.Errorf("create resources error: %v", err)
+	}
+	appc.Resources = res
+
+	return appc, nil
+}
+
+func (c *HAProxy) SidecarContainers(spec *api.PodSpec, secrets string) ([]corev1.Container, error) {
+	res, err := app.CreateResources(spec.SidecarResources)
+	if err != nil {
+		return nil, fmt.Errorf("create sidecar resources error: %v", err)
+	}
+	return []corev1.Container{
+		{
+			Name:            "pxc-monit",
+			Image:           spec.Image,
+			ImagePullPolicy: corev1.PullAlways,
+			Args: []string{
+				"/usr/bin/peer-list",
+				"-on-change=/usr/bin/add_pxc_nodes.sh",
+				"-service=$(PXC_SERVICE)",
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name:  "PXC_SERVICE",
+					Value: c.labels["app.kubernetes.io/instance"] + "-" + "pxc",
+				},
+				{
+					Name: "MONITOR_PASSWORD",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: app.SecretKeySelector("my-cluster-secrets", "monitor"),
+					},
+				},
+			},
+			Resources:              res,
+			TerminationMessagePath: "/dev/termination-log",
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "haproxy-cfg",
+					MountPath: "/etc/haproxy/pxc",
+				},
+			},
+			SecurityContext: &corev1.SecurityContext{
+				Capabilities: &corev1.Capabilities{
+					Add: []corev1.Capability{"SYS_PTRACE"},
+				},
+			},
+		},
+	}, nil
+}
+
+func (c *HAProxy) PMMContainer(spec *api.PMMSpec, secrets string, cr *api.PerconaXtraDBCluster) (corev1.Container, error) {
+	return corev1.Container{}, nil
+}
+
+func (c *HAProxy) Volumes(podSpec *api.PodSpec, cr *api.PerconaXtraDBCluster) (*api.Volume, error) {
+	vol := app.Volumes(podSpec, haproxyDataVolumeName)
+	vol.Volumes = append(
+		vol.Volumes,
+		app.GetConfigVolumes("haproxy-auto", "haproxy-auto"),
+		corev1.Volume{
+			Name: "haproxy-cfg",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		})
```

Comment:
there are two issues
1) I think `haproxy-auto` volume should be temporary, not haproxy-cfg
2) `haproxy-cfg` should mount ConfigMap with name `ls["app.kubernetes.io/instance"]+"-"+ls["app.kubernetes.io/component"]` (see volumes in `pkg/pxc/app/statefulset/node.go`)

Reply:
@hors

Reply:
@delgod this is correct way it is configured in the haproxy to work

Reply:
@hors do we really name "auto" config which customized by used?

Reply:
@delgod  yes :( it was copy past from pxc but I understand that in pxc it is really 'auto' but customer can edit it too )))
So we can rename it e.g. to haproxy-global.cfg  and add the following changes to statefulset:

```
@@ -68,8 +68,8 @@
         volumeMounts:
         - mountPath: /etc/haproxy/pxc
           name: haproxy-cfg
-        - mountPath: /etc/haproxy-auto/
-          name: haproxy-auto
+        - mountPath: /etc/haproxy-custom/
+          name: some-name-haproxy
       - args:
         - /usr/bin/peer-list
         - -on-change=/usr/bin/add_pxc_nodes.sh
@@ -98,9 +98,9 @@
       volumes:
         - name: haproxy-cfg
           emptyDir: {}
-        - name: haproxy-auto
+        - name: some-name-haproxy
           configMap:
-            name: haproxy-auto
+            name: some-name-haproxy
             defaultMode: 420
             optional: true
   updateStrategy:
```
@delgod  what do you think?

Reply:
@hors I would say
```
@@ -68,8 +68,8 @@
         volumeMounts:
         - mountPath: /etc/haproxy/pxc
-          name: haproxy-cfg
+          name: haproxy-auto
-        - mountPath: /etc/haproxy-auto/
-          name: haproxy-auto
+        - mountPath: /etc/haproxy-custom/
+          name: haproxy-custom
       - args:
         - /usr/bin/peer-list
         - -on-change=/usr/bin/add_pxc_nodes.sh
@@ -91,6 +91,8 @@
         volumeMounts:
         - mountPath: /etc/haproxy/pxc
-          name: haproxy-cfg
+          name: haproxy-auto
+        - mountPath: /etc/haproxy-custom/
+          name: haproxy-custom
         securityContext:
           capabilities:
             add:
@@ -98,9 +100,9 @@
       volumes:
-        - name: haproxy-cfg
+        - name: haproxy-auto
           emptyDir: {}
-        - name: haproxy-auto
+        - name: haproxy-custom
           configMap:
-            name: haproxy-auto
+            name: some-name-haproxy
             defaultMode: 420
             optional: true
   updateStrategy:
```